### DISCUSSION TITLE
Whatever video you download, the quality is not right, you will not see it.

### DISCUSSION THREAD

**Discussion by Alpha02S** [aclap-dev/video-downloadhelper#984] [Help and Q&A] [answered]
### Your question

 Hello, I hope you are well. Today I have taken your premium plan. Whatever video quality you download, it doesn't show. Please find a solution for this. I have mailed you but no reply from youPlease answer me as soon as possible, I am waiting for your reply.  Thank you 

### Addon & CoApp version and details

 Hello, I hope you are well. Today I have taken your premium plan. Whatever video quality you download, it doesn't show. Please find a solution for this. I have mailed you but no reply from youPlease answer me as soon as possible, I am waiting for your reply.  Thank you

**Comment by jcfrog** [comment, marked as answer]
Hi,
If you want help you need to give more information like your versions, and a link for a public video as an example to be tested.

**Reply by Alpha02S** [reply]
Thanks a lot for replying. I have downloaded more videos but none of them are playing Here is the video link. 
 "https://www.udemy.com/assets/56039002/encrypted-files/out/v1/62cf96ef545a4ab2802a93077af39f1a/3e691b0da4414b83836363226ac161e9/ecd8798041b7474bb05daed9aae97b93/index.mpd?token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoib3V0L3YxLzYyY2Y5NmVmNTQ1YTRhYjI4MDJhOTMwNzdhZjM5ZjFhLzNlNjkxYjBkYTQ0MTRiODM4MzYzNjMyMjZhYzE2MWU5L2VjZDg3OTgwNDFiNzQ3NGJiMDVkYWVkOWFhZTk3YjkzLyIsImV4cCI6MTcyMDUzOTE0NH0.D_V4W9XtFDivwGYqEQAtkDBbRjB7zZS68vIj75Ic_84&provider=cloudfront&v=1&CMCD=ot%3Dm%2Csf%3Dd%2Csid%3D%220920f32f-7e3a-48b2-9a19-34604e9ef704%22%2Csu" 

Please find a solution for this, it is important. Thanks

Here is Versions

version: 9.0.2.12
target: google
channel: stable
lang: en-US
coapp: {"found":true,"new_version":false,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19"}
license: {"accepted":true,"email":"<removed>","key":"e4026f3f69ce47ea"}
platform: x86-32 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/126.0.0.0 Safari/537.36
{
    "valid_license_message_has_been_displayed": true,
    "used_sidebar_button": true,
    "used_popup_button": true,
    "last_download_directory": {
        "__serializer_tag": "option_some",
        "__serializer_value": {
            "__serializer_tag": "primitive",
            "__serializer_value": "C:\\Users\\Kashif\\Documents"
        }
    },
    "concurrent_downloads_max": 10,
    "never_show_no_incognito_msg_again": true,
    "view_options": {
        "all_tabs": false,
        "hide_downloaded": false,
        "low_quality": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "sort_by_status": true,
        "sort_reverse": false
    },
    "use_sidebar": false,
    "open_count_store": 9,
    "successfull_dl": 35
}

**Comment by Alpha02S** [comment]
Please reply as soon as possible as my Udemy plan is going to expire soon.

**Comment by Mattmjs** [comment]
That link is an indication it is protected and it is indeed with DRM. It's mostly a black screen with no sound.
There is some colors that come and go, that shows that it is a scrambled useless file. Nothing can be done about that.
Udemy is protecting the content.

![udemy](https://github.com/aclap-dev/video-downloadhelper/assets/160297904/23e9b18c-3ba6-47e0-99a9-526fd8d91a52)

**Reply by Alpha02S** [reply]
I believe it will be the same as you said but I downloaded Udemy video it when I used DownloadHelper for free. When i use premium download helper it is downloading but as you said black screen with no sound. 
Please tell me something, I have to download the course from there. If there is any solution please help me.
I paid money to Downloadhelper so that I can download the course from there.  If it can't be downloaded and I opted for the Download Helper premium plan today, please get my money back.  Please help me

**Reply by Alpha02S** [reply]
I request from download helper team that if they give me a positive answer, I will wait for your reply. thanks

**Reply by Mattmjs** [reply]
I'm a user of the software like you but I have to say that is pure coincidence your theory. I believe some things you got were
not protected why others were protected. You got lucky on those videos that were ok. The license plays no part in that.
The license only removes the 120 minute limit. And there is no solution not just in VDH but all software. I have no knowledge of
other software breaking DRM. You can request a refund.

**Reply by Alpha02S** [reply]
Thank you very much I want to get my money back tell me what to do and I'm not too late, I took it a few hours ago.

**Comment by Alpha02S** [comment]
Can someone tell me what I need to do to get the money back? I have been trying for the last 2 hours but no response. please

**Reply by GrampaWildWilly** [reply]
I do not work for VDH.  I'm just another user.

First, you need to remember that @jcfrog Jérôme lives in France.  He's the VDH employee who handles all license issues.  Take the difference in time zones into account when you start getting anxious about not hearing any answer in what you define as a timely manner.  You might be expecting a response at a time when Jérôme is asleep.  24x7 support is not available here.  You need to lower your expectations.

Second, you are asking for a refund in the wrong place.  Go back to the home page of this forum.  The link to that page is at the top of this page.  The link says `Discussions`.  Click that.  That takes you to the home page of this forum.  Then look at the top of the page.  There are some large squares with links in them.  These are pinned help topics.  Click the one about licenses.  You will find further instructions there.

**Reply by Alpha02S** [reply]
Your way of explaining was very good but the issue was resolved at the same time they sent a mail saying they will refund the money. Thank you very much @jcfrog Jérôme.

**Reply by GrampaWildWilly** [reply]
Just a comment on the original issue.  Udemy and Hotmart have been hot topics in our forum, both here on GitHub recently & in the past on our old forum on Google Groups.  We've gotten conflicting reports.  Some people have no problems downloading with VDH & others can't do it.  It seems that different courses exhibit different properties.  Some courses are protected by DRM & some are not.  Even more confusing, some parts of some courses are protected by DRM & other parts of the same courses are not.  So you can download part of a course but not all of it.  There doesn't seem to be any consistency on these web sites.  I suppose it's possible that you could ask the author of any given course to remove the DRM protection.  My personal take is that you subscribers paid for the course.  The content is not universally available because it's behind the site's paywall.  That ought to be protection enough.  In the old days, many decades ago when I was in school, you bought the textbook & you could read it in bed, on the bus, in the library, on the toilet, anywhere, any time.  Downloading a course from one of these education sites is merely the modern day equivalent of holding the textbook in your hands.  To me, this whole DRM thing is just wrongheaded.  There should be no DRM protections on these courses.  Maybe such an argument might sway some individual course author.  Perhaps you could start a movement that would change a site-wide policy.  I wish you good luck.
